### ISSUE TITLE
Error when using this library

### ISSUE THREAD

**Issue by WayneWeiZhang** [plausiblelabs/plweakcompatibility#7]
Hi,
When I copied PLWeakCompatibilityStubs into my project, I got error which show me "Conflicting type for 'objc_loadWeak' and similar Conflicting type for 'objc_storeWeak'".
I wonder how can I solve this error.
Thanks!

**Comment by mikeash** [comment]
I'm guessing that you have a prefix header which includes (probably indirectly) the ObjC runtime headers, and this is conflicting with the definitions in this code. Try turning off the prefix header in your build settings (I don't find them to be very clean in general, since they end up making code magically appear at the top of your files without any indication that this is happening) and see if that remedies the problem. You may need to fix up some missing #imports in other files if they depend on the prefix header, but I think that's ultimately a good thing.

**Comment by WayneWeiZhang** [comment]
Thanks, mikeash!
Already fixed it

**Comment by evadne** [comment]
I ran into the same issue. Apparently, if you attempt to use `<objc/runtime.h>` anywhere in the same target, this won’t fly with Xcode. My solution for people who need to target a flat `xcodeproj` is to simply compile `PLWeakCompatibility` separately as a static library then link it into your app:
- Specify weak references to be available at runtime. In Other C Flags, add `-Xclang -fobjc-runtime-has-weak`.
- Allow the linker to pick up the stubs. In Linker Flags, add `-ObjC`.
- Include `PLWeakCompatibility.xcodeproj` directly into your application project.
- In Build Phases → Dependencies, add PLWeakCompatibility-iOS.
- In Build Phases → Link Binary With Libraries, add libPLWeakCompatibility-iOS.
- In your Scheme → Build → Your Application, make sure PLWeakCompatibility-iOS is there under your app target.

Basically don’t mix it with other code lest the linker complain.
